packages/hash/shared/src/queries/page.queries.ts

```diff
@@ -1,5 +1,40 @@
 import { gql } from "@apollo/client";
 
+const persistedBlockFieldsFragment = gql`
+  fragment PersistedBlockFields on PersistedBlock {
+    __typename
+    entityId
+    entityVersion
+    accountId
+    entityTypeId
+    componentId
```

Comment:
I think `componentId` should be in `properties` – a block is an entity like any other, which in its pure BP representation should look like
```
{
  entityId
  entityTypeId
  properties {
    componentId
  }
}
```
with a separate link to its child entity

Reply:
It is available in the properties, but the key would be its base URI so that query would not work. The top-level `componentId` field resolver acts like an alias for that property.

Reply:
Ah right, I understand the motivation now, I wasn't thinking about the fact that this was now a URI.

I don't think adding a field resolver when we want to access properties on a type defined in the workspace API is a good pattern to establish. It adds unnecessary overhead, and makes the presence of the `properties` blob pointless since we're never actually using it. It also means that the FE needs to know to translate `componentId` back into its URI when sending a properties _update_, or (even worse in my view) we need to then add special handlers for every property when they are updated to allow the FE to use this shorter form.

Reply:
It also makes it unclear which of these things are truly top-level fields on an entity, which are special handlers for other entities (`dataEntity`), and which are shortcuts to things in the `properties` blob.

The amount of special casing and 'this thing isn't really there but we pretend it is' makes the current system very confusing, and I'd rather we kept it to a minimum in the new one.

Reply:
([reference to other comment](https://github.com/hashintel/hash/pull/1196/files/4db20e72149725297f5560645c1907d4ca862769#r997153374) this point applies to – I've resolved it to keep discussion in one place)

Reply:
This is currently done for workspace types as we have full control, but I see how it could be confusing/misleading when dealing with `block`s as the API is somewhat exposed to the underlying "components". We could move this back out to properties and make use of the ontology URIs from the shared project to access the property field, but I'm not sure we would gain much if this code is going to be thrown away relatively soon.

I think it's worth it to make the change for new collab, though, as we do have the opportunity to make things more consistent.

Reply:
Sounds good – presumably the FE will be going through collab anyway, and collab will be the consumer of these types.

Reply:
Just for clarification, do you want me to move this out to `properties` again, or should we skip this?